code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -84,6 +123,25 @@
 			new /obj/item/reagent_containers/glass/waterbottle/relic(src)
 		if(30)
 			new /obj/item/reagent_containers/glass/bottle/necropolis_seed(src)
+	if(prob(10)) // Low chance to give you a hat, funny tf2
+		var/hat = pick(list(
+			/obj/item/clothing/head/beanie,
+			/obj/item/clothing/head/helmet/gladiator,
+			/obj/item/clothing/head/flatcap,
+			/obj/item/clothing/head/pirate,
+			/obj/item/clothing/head/beret,
+			/obj/item/clothing/head/plaguedoctorhat,
+			/obj/item/clothing/head/ushanka,
+			/obj/item/clothing/head/bowler,
+			/obj/item/clothing/head/that,
+			/obj/item/clothing/head/bandana,
+			/obj/item/clothing/head/fedora,
+			/obj/item/clothing/head/chefhat,
+			/obj/item/clothing/head/hardhat,
+			/obj/item/clothing/head/crown,
+			/obj/item/clothing/head/festive
+			))
+		new hat(src)
```

Comment:
Standard way to do this would be creating a static list; static lists will be defined only once; i.e. list creation will happen only once.
```suggestion
		var/static/list/necropolis_hat_drops = list(
			/obj/item/clothing/head/beanie,
			/obj/item/clothing/head/helmet/gladiator,
			/obj/item/clothing/head/flatcap,
			/obj/item/clothing/head/pirate,
			/obj/item/clothing/head/beret,
			/obj/item/clothing/head/plaguedoctorhat,
			/obj/item/clothing/head/ushanka,
			/obj/item/clothing/head/bowler,
			/obj/item/clothing/head/that,
			/obj/item/clothing/head/bandana,
			/obj/item/clothing/head/fedora,
			/obj/item/clothing/head/chefhat,
			/obj/item/clothing/head/hardhat,
			/obj/item/clothing/head/crown,
			/obj/item/clothing/head/festive
			)
		var/hat = pick(necropolis_hat_drops)
		new hat(src)
```

Reply:
Typecacheof too?

Reply:
a simple list of typepaths will do.